wire/wire_io.c

```diff
@@ -32,9 +32,13 @@ static int do_read_wire_header(int fd, struct io_plan_arg *arg)
 		return -1;
 	arg->u2.s += ret;
 
-	/* Both bytes read?  Set up for normal read of data. */
+	/* Length bytes read?  Set up for normal read of data. */
 	if (arg->u2.s == INSIDE_HEADER_BIT + HEADER_LEN) {
-		arg->u2.s = be16_to_cpu(*(be16 *)p);
+		arg->u2.s = *(wire_len_t *)p;
```

Comment:
Ever since implementing the Bitcoin protocol I have a very strong aversion against using native host byte order for anything that could potentially be used in a networked scenario. I know we currently don't send this anywhere but the same host, but I'd use network byte order anyway.

Reply:
Yes, was a line-ball for me.  I'll fix.